### ISSUE TITLE
why the use of http.HandlerFunc

### ISSUE THREAD

**Issue by rickb777** [urfave/negroni#174]
Why does Negroni use both `http.Handler` and `http.HandlerFunc` in its API? It's rather inconsistent.

```
func New(handlers ...Handler) *Negroni
```
compared with the third parameter in
```
type Handler interface {
	ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc)
}
```

**Comment by dineshkumar-cse** [comment]
negroni.Handler interface will have 
```
  ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.Handler)
```
instead of `http.HandlerFunc`.

and we 'll need a new release. But this will affect the consumers when they upgrade.
I can change it and raise a PR if owners/maintainers give it a go.

**Comment by jszwedko** [comment]
I agree that it is a little inconsistent, but I hesitate to modify this part of the API until a version `2` given the severity of the breaking change.

**Comment by Solaris-star** [comment]
The `next http.HandlerFunc` parameter in the `Handler` interface is intentional — it's the closure that calls the next middleware in the chain. Using `http.HandlerFunc` (a concrete type) rather than `http.Handler` (an interface) is correct here because `next` is always a function, never a struct implementing `ServeHTTP`.

This follows the same pattern as `net/http`'s middleware wrapping. The API is consistent: you pass `http.Handler` values to `New()`, and the chain wires them via `http.HandlerFunc` closures.
